Migrate to pyproject.toml

RishiP2006 opened the pull request:
This PR migrates the project to use a pyproject.toml file for configuration and packaging instead of the previous requirements.txt system. This change standardizes the build process, simplifies dependency management, and aligns the project with modern Python packaging standards. The migration addresses issue #43 .

Changes include:

Creation of a pyproject.toml file containing project metadata, dependencies, and build-system configuration.
Removal of legacy files (requirements.txt).
Updated documentation to reflect the new configuration.
Dependencies added in pyproject.toml:

nowcasting_datamodel==1.5.56
numpy
pandas
freezegun
pytest
testcontainers
sentry-sdk==2.13.0
How Has This Been Tested?
Installed the package in editable mode with pip install -e ..

Ran the full test suite using pytest (with PYTHONPATH set appropriately) and all tests passed.

RishiP2006 commented:
Note on Warnings:
While running tests, some warnings (e.g., deprecation warnings from Pydantic and FutureWarnings from Pandas) were displayed. These warnings stem from third-party libraries and do not impact the core functionality of the migration.

RishiP2006 commented:
Hi @peterdudfield, 

I hope I can get some feedback on it, anything that I have to improve?

peterdudfield commented on build/lib/forecast_blend/__init__.py:
```diff
@@ -0,0 +1 @@
+""" For blending forecasts """
```
can you not add this folder please

peterdudfield commented on forecast_blend.egg-info/PKG-INFO:
```diff
@@ -0,0 +1,90 @@
+Metadata-Version: 2.4
```
can you remove this folder please

peterdudfield commented:
And please can requirements.txt be removed, if its not needed anymore

peterdudfield commented:
Would you be able to change the author to Open Climate Fix?

peterdudfield commented on forecast_blend/__pycache__/__init__.cpython-312.pyc:
These files still need to be removed

peterdudfield commented on pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+requires-python = ">=3.8"
+license = {text = "MIT License"}
+authors = [
+    { name = "Rishi Prasad", email = "rishiprasad002@gmail.com" }
```
sorry, could you take your name out, and just write Open Climate Fix,with info@openclimatefix.org

peterdudfield commented on tests/__pycache__/__init__.cpython-312.pyc:
These files needs to be removed

RishiP2006 commented:
@peterdudfield I removed the folders and changed the author information, please check and tell me anything else that needs to be done.

Thanks, 
Rishi

peterdudfield commented:
There are still some .pyc files that need removing

peterdudfield commented:
Please remove the .pyc files. A way to see this is to look at https://github.com/openclimatefix/uk-pv-forecast-blend/pull/45/files
